src/sentry/static/sentry/app/data/forms/projectDebugFiles.jsx

```diff
@@ -1,8 +1,45 @@
+import _ from 'lodash';
+import React from 'react';
+
 import {t} from 'app/locale';
+import {openDebugFileSourceModal} from 'app/actionCreators/modal';
+import TextBlock from 'app/views/settings/components/text/textBlock';
 
 // Export route to make these forms searchable by label/help
 export const route = '/settings/:orgId/projects/:projectId/debug-symbols/';
 
+export const sourceNames = {
+  gcs: t('Google Cloud Storage'),
+  http: t('SymbolServer (HTTP)'),
+  s3: t('Amazon S3'),
+};
+
+export function getSourceName(type) {
+  return sourceNames[type] || t('Unknown');
+}
+
+function flattenKeys(obj) {
+  const result = {};
+  _.forEach(obj, (value, key) => {
+    if (_.isObject(value)) {
+      _.forEach(value, (innerValue, innerKey) => {
```

Comment:
Do you ever need to recursively flatten keys beyond 2 levels?

Reply:
No, we only have one level of nesting in these objects. 

To be fair, this is just a hack because our form models cannot deal with nested objects. Ideally, I would like to get rid of this completely.